lib/itests/core/src/test/java/io/atlasmap/itests/core/NestedCollectionJavaTest.java

```diff
@@ -155,6 +155,37 @@ public void testAsymmetricPaths2ndLevelNestedCollectionTo4LevelsNestedCollection
         assert2ndLevelCollection(nestedTarget);
     }
 
+    @Test
+    public void test3To2LevelNestedCollection() throws Exception {
+        TargetClass target = processNestedJavaCollection(Arrays.asList("3-2"));
+
+        assertEquals(2, target.getSomeArray().length);
+        assertEquals(4, target.getSomeArray()[0].getSomeArray().length);
+        assertEquals("array000", target.getSomeArray()[0].getSomeArray()[0].getSomeField());
+        assertEquals("array001", target.getSomeArray()[0].getSomeArray()[1].getSomeField());
+        assertEquals("array002", target.getSomeArray()[0].getSomeArray()[2].getSomeField());
+        assertNull(target.getSomeArray()[0].getSomeArray()[3].getSomeField());
```

Comment:
Not sure about the behavior here. It's different from JSON and XML, but seems to have been added explicitly. It basically includes items as null values for paths when some child levels are missing in source and adds a warning for those paths. See https://github.com/atlasmap/atlasmap/blob/master/lib/modules/java/core/src/main/java/io/atlasmap/java/core/JavaFieldReader.java#L104-L109 For JSON and XML such paths are simply skipped.

Reply:
These are the affected tests when I remove that functionality and make it the same as json and xml: https://github.com/atlasmap/atlasmap/blob/master/lib/modules/java/core/src/test/java/io/atlasmap/java/core/JavaFieldReaderTest.java#L93-L128

Reply:
Hmm... say Company<>/Employee<>/Contact/Email, and some employee could have null contact. From that perspective Java reader behavior might be better. We might have to change json/xml behavior instead? It would be a separate bug.

Reply:
I'm sorry I'll respond tomorrow as I'm too tired to reason about it today :)

Reply:
You are right! The Java reader behavior is better. The logic would be otherwise broken if one mapped 2 fields e.g. Company<>/Employee<>/Contact/Email and Company<>/Employee<>/Contact/Name. If we skip one Employee due to null Contact or Email then we will adjust indexes after that employee and the next mapping will assign wrong names for employees for adjusted indexes. It's just a bit awkward to see null entries in a merged collection when source collection count > target collection count. Maybe in such a case (since indexes are not preserved anyway) we should skip null values. I'll create a separate issue to fix these.

Reply:
Added https://github.com/atlasmap/atlasmap/issues/1906